ksy_schema.json

```diff
@@ -241,6 +241,10 @@
         "encoding": {
           "type": "string"
         },
+        "pad-right": {
+          "type": "integer",
```

Comment:
So this is not an issue specific to this PR, and I don't think it makes sense to fix it in this PR, I just want to point it out: For `pad-right`, `terminator`, etc. the type `"integer"` is not quite correct - KS allows not only native YAML integers, but also strings with integers in them, so that for example hex literals can be used. The spec currently only accepts native YAML integers, so `pad-right: 0x20` is incorrectly treated as an error.

Reply:
You are right - on top of that, you can even of course use expressions to calculate and involve the sequence field values that you have read previously, e. g. `pas

Reply:
> on top of that, you can even of course use expressions to calculate and involve the sequence field values that you have read previously

Arbitrary expressions aren't allowed for `terminator` and `pad-right`, only literal integer constants. For example, ksc accepts `0x20`, but not `0x20 + 0`. `StringOrInteger` is still a good solution though.

In the future, these fields could be specified more precisely using a regex pattern that matches all possible integer literals. But for now `StringOrInteger` is enough I think.

Reply:
Have you tested that? I tried `terminator: '0x40'` (forcing string interpretation) in web ide and got: `io.kaitai.struct.format.YAMLParseException: /seq/0/terminator: expected int, got Some(0x40) (class scala.Some)`.

I looked into YAML specs and it allows various formats of specifying integers:

```
canonical: 12345
decimal: +12345
octal: 0o14
hexadecimal: 0xC
```

So specifying type: "integer" reflects the current compiler behavior.

Reply:
You're right - `0x20` is actually a valid YAML integer, and `"0x20"` is not accepted by ksc, so the type `integer` is correct here.

It seems that the IntelliJ plugin for JSON Schema validation is buggy - it incorrectly detects hex and octal literals as strings instead of integers, so it reported a schema validation error for `pad-right: 0x20`, which led me to think that ksc accepts strings here (even though it actually doesn't).

Reply:
Actually, I don't see why we should not allow arbitrary expressions in `terminator` and `pad-right`.